cargo-concordium/src/build.rs

```diff
@@ -375,6 +381,291 @@ pub fn init_concordium_project(path: impl AsRef<Path>) -> anyhow::Result<()> {
+    }
+
+    // save the schema JSON representation into the file.
+    absolute_path_of_out.push(contract_name.to_owned() + "_schema.json");
+
+    File::create(absolute_path_of_out.clone())?;
+
+    std::fs::write(absolute_path_of_out, serde_json::to_string(&schema_json)?)?;
+
+    Ok(())
+}
+
+/// Converts the ContractV3 schema of the given contract_name to JSON and writes
+/// it to a file named after the smart contract name at the specified location.
+pub fn write_json_schema_to_file_v3(
+    mut absolute_path_of_out: PathBuf,
+    contract_name: &String,
+    contract_schema: &ContractV3,
+) -> anyhow::Result<()> {
+    // create empty schema_json
+    let mut schema_json: Value = serde_json::from_str(r#"{}"#)?;
+
+    // add init schema
+    if let Some(init_schema) = &contract_schema.init {
+        // create empty init_function_object
+        let mut init_function_object: Value = serde_json::from_str(r#"{}"#)?;
+
+        // add parameter schema to init_function_object
+        if let Some(parameter_schema) = &init_schema.parameter {
+            init_function_object["parameter"] =
+                json!(general_purpose::STANDARD.encode(to_bytes(parameter_schema)))
+        }
+
+        // add return_value schema to init_function_object
+        if let Some(return_value_schema) = &init_schema.return_value {
+            init_function_object["return_value"] =
+                json!(general_purpose::STANDARD.encode(to_bytes(return_value_schema)))
+        }
+
+        // add error schema to init_function_object
+        if let Some(error_schema) = &init_schema.error {
+            init_function_object["error"] =
+                json!(general_purpose::STANDARD.encode(to_bytes(error_schema)))
+        }
+
+        schema_json["init"] = init_function_object;
+    }
+
+    // add event schema
+    if let Some(event_schema) = &contract_schema.event {
+        schema_json["event"] = json!(general_purpose::STANDARD.encode(to_bytes(event_schema)))
+    }
+
+    // add receive entrypoints
+    if !contract_schema.receive.is_empty() {
+        // create empty entrypoints
+        let mut entrypoints: Value = serde_json::from_str(r#"{}"#)?;
+
+        // iterate through the entrypoints and add their schemas
+        for (method_name, receive_schema) in contract_schema.receive.iter() {
+            // create empty receive_function_object
+            let mut receive_function_object: Value = serde_json::from_str(r#"{}"#)?;
+
+            // add parameter schema to receive_function_object
+            if let Some(parameter_schema) = &receive_schema.parameter {
+                receive_function_object["parameter"] =
+                    json!(general_purpose::STANDARD.encode(to_bytes(parameter_schema)))
+            }
+
+            // add return_value schema to receive_function_object
+            if let Some(return_value_schema) = &receive_schema.return_value {
+                receive_function_object["return_value"] =
```

Comment:
We should use `camelCase` (i.e., `returnValue`) for JSON I think. That's what we do usually.